MedallionShell.Tests/Streams/MergedLinesEnumerableTestBase.cs

```diff
@@ -84,54 +85,69 @@ void TestOneThrows(bool reverse)
                 mockReader.Setup(r => r.ReadLineAsync())
                     .ReturnsAsync(() => ++count < 3 ? "LINE" : throw new TimeZoneNotFoundException());
 
-                Assert.Throws<TimeZoneNotFoundException>(
-                    () => new MergedLinesEnumerable(
+                Assert.ThrowsAsync<TimeZoneNotFoundException>(
+                    async () => await this.Create(
                             reverse ? mockReader.Object : reader1,
                             reverse ? reader1 : mockReader.Object
-                        )
-                        .ToList()
+                        ).ToListAsync()
                 );
             }
 
             TestOneThrows(reverse: false);
             TestOneThrows(reverse: true);
         }
 
+        [Timeout(10000)] // something's wrong if it's taking more than 15 seconds
```

Comment:
Please swap the order of this and the `Test` attribute.

I like `[Test, Timeout(10000)]`